### ISSUE TITLE
SIMPLIFY and MERGE should ignore unhappy intervals

### ISSUE THREAD

**Issue by nickhale** [chebfun/chebfun#990]
Consider the following *, which fails to revolve fully with the default `maxSplitLength`:
`G = chebfun(@(x) mod(1./x,1), [1/1000, 1], 'splitting', 'on');`

( \* This example is taken from Corless' book "A Graduate Introduction to Numerical Methods" - #376. Unfortunately it passed in V4 but no longer passes in V5. You can fix it by setting `maxSplitLength = 1e5`. Note that it passes in V4 more by luck than judgement. Indeed, using the domain [-1, -1/1000] it fails in V4 too. The issue is related to the fact that edge detection is not perfect, and that in V5 or strategy for choosing which subinterval to split has changed. (We now choose the largest unhappy piece. Before we chose the first unhappy piece to the left, which is what allowed this example to work.))

Anyway, you get something like this:

```
Warning: Function not resolved using 6079 pts. 
> In chebfun.constructor>constructorSplit at 262
  In chebfun.constructor at 61
  In chebfun.chebfun>chebfun.chebfun at 219 
G =
   chebfun column (38 smooth pieces)
.
.
.
Epslevel = 5.198208e-02.  Vscale = 1.894693e+00.  Total length = 4280.
```

and if you plot it, you see
![corless1](https://cloud.githubusercontent.com/assets/2035010/3352378/7130d902-fa43-11e3-9852-90c3e08f4270.png)

If you comment out the merge at the end of the `chebfun` constructor, you get 

```
>> G = chebfun(@(x) mod(1./x,1), [1/1000, 1], 'splitting', 'on')
Warning: Function not resolved using 6079 pts. 
> In chebfun.constructor>constructorSplit at 262
  In chebfun.constructor at 61
  In chebfun.chebfun>chebfun.chebfun at 219 
G =
   chebfun column (178 smooth pieces)
.
.
.
Epslevel = 5.198208e-02.  Vscale = 1.000000e+00.  Total length = 6079.
```

and
![corless2](https://cloud.githubusercontent.com/assets/2035010/3352383/968cb0ae-fa43-11e3-9b25-62597ed2d4ec.png)

The latter at least looks reasonable in the intervals where the construction was happy.

Basically, by merging / simplifying we pollute the unhappy intervals with the junk from the unhappy ones.
